en/guide/mavlink_version.md

```diff
@@ -1,27 +1,52 @@
+- [MAVLink 2.0](../guide/mavlink_2.md): Current/recommended major version. Adopted by major users early 2017. 
+- *MAVLink v1.0*: Widely adopted around 2013. Still used by a number of legacy peripherals.
+
+The *MAVLink 2.0* C/C++ and Python libraries are backwards compatible with MAVLink 1.0 (support both protocols).
+[Version Handshaking](#version_handshaking) and [Negotiating Versions](#negotiating_versions) explain how to choose which version is used.
+  
+> **Note** *MAVLink v0.9* is a pre-release version that is no longer supported. 
+  The associated message set deleted in August 2018. 
+  Legacy code may be present in generator and test code. 
+
+
+## Determining Protocol/Message Version
+
+A library's MAVLink support can be determined in a number of ways:
+* [AUTOPILOT_VERSION](../messages/common.md#AUTOPILOT_VERSION)`.capabilities` can be checked against the [MAV_PROTOCOL_CAPABILITY_MAVLINK2](../messages/common.md#MAV_PROTOCOL_CAPABILITY_MAVLINK2) flag to verify MAVLink 2 support.
+* [PROTOCOL_VERSION](../messages/common.md#PROTOCOL_VERSION).`version` contains the MAVLink version number multiplied by 100: v1.0 is 100, <!-- v2.0 is 200, --> v2.3 is 203 etc.
+* [HEARTBEAT](../messages/common.md#HEARTBEAT)`.mavlink_version` field contains the minor version number. 
+  This is the `<version>` field defined in the [Message Definitions](../messages/README.md) (`version` in [common.xml](../messages/common.md) for dialects that depend on the common message set).
+* The major version can be determined from the packet start marker byte: 
+  * MAVLink 1: `0xFE` 
+  * MAVLink 2: `0xFD`
+
+  > **Note** A MAVLink library that does not support a protocol version will not recognise the protocol start marker; 
+    so no messages will be even be detected (see [Serialization](../guide/serialization.md)).
+
+> **Tip** While messages do not contain version information, an extra CRC is used to ensure that a library will only process compatible messages (see [Serialization > CRC_EXTRA](../guide/serialization.md)).
+
 
-*MAVLink v1.0* was widely adopted around 2013. 
-*MAVLink 2.0* was adopted by major users early 2017, but there are still quite a few legacy peripherals in use that don't yet support it. 
-The *MAVLink 2.0* C/C++ and Python libraries are backwards compatible with MAVLink 1.0, which has tremendously simplified the transition.
 
 ## Version Handshaking {#version_handshaking}
 
-Support for *MAVLink 2* is indicated in the [AUTOPILOT\_VERSION](../messages/common.md#AUTOPILOT_VERSION) message by the [MAV_PROTOCOL_CAPABILITY_MAVLINK2](../messages/common.md#MAV_PROTOCOL_CAPABILITY_MAVLINK2) flag. 
+Support for *MAVLink 2* is indicated in the [AUTOPILOT_VERSION](../messages/common.md#AUTOPILOT_VERSION) message by the [MAV_PROTOCOL_CAPABILITY_MAVLINK2](../messages/common.md#MAV_PROTOCOL_CAPABILITY_MAVLINK2) flag. 
+
 This is sufficient if the communication link between autopilot and GCS is completely transparent. 
```

Comment:
@LorenzMeier 

> This is sufficient if the communication link between autopilot and GCS is completely transparent. 

The implication is that this is not sufficient for testing MAVLink 2 support if the link is not transparent. 

I don't think that is what you mean. I think you are saying that (irrespective of MAVLink 2 support) if the link is not transparent then it is worth doing a round-trip handshake to verify the whole path. 
- Is that correct?
- Why is this so? Presumably if you send request and get back the AUTOPILOT_VERSION you know that sending and receiving works?

Reply:
Yes, you need to do a round-trip. And yes, if a MAVLink 2 message came back as MAVLink 2 then you know that the autopilot understands it and the link does support it too.